### PR TITLE
RDBC-532, RDBC-577, RDBC-582

### PR DESCRIPTION
**Pull request by poissoncorp**
https://issues.hibernatingrhinos.com/issue/RDBC-577
https://issues.hibernatingrhinos.com/issue/RDBC-532
https://issues.hibernatingrhinos.com/issue/RDBC-582

### CODE REVIEW

File: ravendb/documents/store/definition.py [hunk 1]
```diff
@@ -126,13 +125,20 @@ def database(self, value: str):
...
-    @certificate_path.setter
-    def certificate_path(self, value: str):
+    @certificate_pem_path.setter
+    def certificate_pem_path(self, value: str):
         self.__assert_not_initialized("certificate_url")
-        self.__certificate_path = value
+        if value.endswith("pfx"):
+            raise ValueError(
+                "Invalid certificate format. "
+                "Please use .pem file or pass tuple with the (.crt path, .key path) values in it."
```
Comment (ml054) [inline ravendb/documents/store/definition.py]: ?

File: ravendb/documents/store/definition.py [hunk 2]
```diff
@@ -126,13 +125,20 @@ def database(self, value: str):
...
+            )
+        if "BEGIN CERTIFICATE" not in open(value).read():
```
Comment (ml054) [inline ravendb/documents/store/definition.py]: check for private as well

File: ravendb/documents/store/definition.py (hunk 1 above)
Comment (poissoncorp) [reply ravendb/documents/store/definition.py]: fixed ;]

File: ravendb/documents/store/definition.py (hunk 2 above)
Comment (poissoncorp) [reply ravendb/documents/store/definition.py]: done

**Pull request merged**